tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -164,6 +164,29 @@ public void TestConstructor_InstanceMetadata_ValidateAuthority_MutuallyExclusive
             Assert.AreEqual(ex.ErrorCode, MsalError.ValidateAuthorityOrCustomMetadata);
         }
 
+        [TestMethod]
+        public void TestConstructor_InstanceMetadataUri_ValidateAuthority_MutuallyExclusive()
+        {
+            var ex = AssertException.Throws<MsalClientException>(() => PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                  .WithInstanceDicoveryMetadata(new Uri("https://some_uri.com"))
+                                                  .WithAuthority("https://some.authority/bogus/", true)
+                                                  .Build());
+            Assert.AreEqual(ex.ErrorCode, MsalError.ValidateAuthorityOrCustomMetadata);
+        }
+
+        [TestMethod]
+        [DeploymentItem(@"Resources\CustomInstanceMetadata.json")]
+        public void TestConstructor_WithInstanceDicoveryMetadata_OnlyOneOverload()
+        {
+            string instanceMetadataJson = File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("CustomInstanceMetadata.json"));
+            var ex = AssertException.Throws<MsalClientException>(() => PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                  .WithInstanceDicoveryMetadata(instanceMetadataJson)
```

Comment:
is there a way to through an error earlier? like when building?

Reply:
This error is thrown when the `Build()` call is made. The builder pattern used here has a `Validate()` method https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/4d2ee65290f88205e6451dddd23695e52a30c872/src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs#L369